### ISSUE TITLE
Infinite Loop - Sign in with Facebook

### ISSUE THREAD

**Issue by bconn98** [sharkwouter/minigalaxy#388]
Running pure arch, I am unable to use the sign in with Facebook option. When I select it, I get an infinite spinning wheel.

**Comment by sharkwouter** [comment]
I haven't been able to test this before the current release. I hope it's not in version 1.1.0, but that would need to be tested.

**Comment by bconn98** [comment]
This is still a concern. It may be an issue beyond your implementation. Lutris crashes when I select the "Sign in with Facebook" button.

**Comment by bconn98** [comment]
Still present

**Comment by phlash** [comment]
The below patch fixes this for me - it's against 1.0.2 but should be very similar for later codebases. Apologies for some fugly butchery to stash the popup window reference in the WebView widget so I can close it automatically, I am not familiar with PyGtk :disappointed: 

```diff
diff --git a/minigalaxy/ui/login.py b/minigalaxy/ui/login.py
index 4dd42a5..a9a31bf 100644
--- a/minigalaxy/ui/login.py
+++ b/minigalaxy/ui/login.py
@@ -24,10 +24,13 @@ class Login(Gtk.Dialog):
 
         self.redirect_url = redirect_url
 
-        context = WebKit2.WebContext.new()
-        webview = WebKit2.WebView.new_with_context(context)
+        # https://stackoverflow.com/questions/9147875/webview-dont-display-javascript-windows-open
+        settings = WebKit2.Settings.new()
+        settings.props.javascript_can_open_windows_automatically = True
+        webview = WebKit2.WebView.new_with_settings(settings)
         webview.load_uri(login_url)
         webview.connect('load-changed', self.on_navigation)
+        webview.connect('create', self.on_create)
 
         self.box.pack_start(webview, True, True, 0)
         self.show_all()
@@ -40,6 +43,24 @@ class Login(Gtk.Dialog):
                 self.result = self.__get_code_from_url(uri)
                 self.hide()
 
+    # Create any pop-up windows during authentication
+    def on_create(self, widget, action):
+        popup = Gtk.Dialog(title=_("Facebook Login"), parent=self, flags=0, buttons=())
+        webview = WebKit2.WebView.new_with_related_view(widget)
+        webview.load_uri(action.get_request().get_uri())
+        webview.__dict__['popup'] = popup
+        webview.connect('close', self.on_close_popup)
+        popup.get_content_area().pack_start(webview, True, True, 0)
+        popup.set_size_request(400, 600)
+        popup.set_modal(True)
+        popup.show_all()
+        return webview
+
+    # When a pop up is closed (by Javascript), close the Gtk window too
+    def on_close_popup(self, widget):
+        if 'popup' in widget.__dict__:
+            widget.__dict__['popup'].hide()
+
     # Return the code when can be used by the API to authenticate
     def get_result(self):
         return self.result
```

**Comment by sharkwouter** [comment]
I've merged your patch, it seems to work well.
